File: lavcode-node/src/actions/password/dtos/update-password.dto.ts
import { V } from "@ipare/validator";
import { UpsertIconDto } from "../../../dtos/upsert-icon.dto";
import { UpsertKeyValuePair } from "./upsert-key-value-pair.dto";

export class UpdatePasswordDto {
  @V().Required()
  folderId!: string;
  @V().IsOptional()
  title?: string;
  @V().IsOptional()
  value?: string;
  @V().IsOptional()
  remark?: string;
  @V().Required()
  order!: string;

  @V().Required()
  icon!: UpsertIconDto;
  @V().Required().Items(UpsertKeyValuePair)
  keyValuePairs!: UpsertKeyValuePair[];
}

File: src/Lavcode.Asp/Dtos/UpdatePasswordDto.cs
using System.ComponentModel.DataAnnotations;

namespace Lavcode.Asp.Dtos
{
    public class UpdatePasswordDto
    {
        [Required]
        public string FolderId { get; set; } = null!;

        public string? Title { get; set; }
        public string? Value { get; set; }
        public string? Remark { get; set; }

        [Required]
        public int Order { get; set; }

        public UpsertIconDto? Icon { get; set; } = null!;
        public UpsertKeyValuePairDto[]? KeyValuePairs { get; set; }
    }
}
